nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/text/TextLineReader.java

```diff
@@ -0,0 +1,125 @@
+public class TextLineReader extends SchemaRegistryService implements RecordReaderFactory {
+
+    private volatile ConfigurationContext context;
+    private volatile String charSet;
+
+    public static final PropertyDescriptor LINE_SEPARATOR = new PropertyDescriptor.Builder()
+            .name("linereader-line-separator")
+            .displayName("Line Separator")
+            .description("Specifies the character(s) to use in order to separate individual lines of text.")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .defaultValue("\\n")
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor LINES_PER_RECORD = new PropertyDescriptor.Builder()
+            .name("linereader-lines-per-record")
+            .displayName("Lines Per Record")
+            .description("The number of lines that will be added to the field's value. This allows groups of lines to be treated as a single string for the purposes of populating the field value. "
+                    + "To split each line into its own record, set this value to 1. A value of zero (0) indicates the entire content is to be treated as a single record.")
+            .required(true)
+            .defaultValue("1")
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor SKIP_LINE_COUNT = new PropertyDescriptor.Builder()
+            .name("linereader-skip-count")
+            .displayName("Skip Line Count")
+            .description("The number of lines at the beginning of the input that should be ignored and thus not treated as records.")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .defaultValue("0")
+            .build();
+
+    protected static final PropertyDescriptor LINE_FIELD_NAME = new PropertyDescriptor.Builder()
+            .name("linereader-field-name")
+            .displayName("Output Field Name")
+            .description("Specifies the name of the record field in the output schema, whose values will be the text lines from the input as configured in the processor.")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    protected static final PropertyDescriptor IGNORE_EMPTY_LINES = new PropertyDescriptor.Builder()
+            .name("linereader-ignore-empty-lines")
+            .displayName("Ignore Empty Lines")
```

Comment:
I'm not sure I understand this. Can you reword and clarify how this relates to other properties?